scripts/pages/settings/speedometer.js

```diff
@@ -218,7 +218,9 @@ class Speedometers {
 		UiToolkitAPI.ShowCustomLayoutPopupParameters(
 			'',
 			'file://{resources}/layout/modals/popups/speedometer-select.xml',
-			`callback=${UiToolkitAPI.RegisterJSCallback((type, name) => this.addSpeedometerByType(type, name))}`
+			`speedometerNames=${this.detailObjectList.map((x) => x.name)}&callback=${UiToolkitAPI.RegisterJSCallback(
```

Comment:
Im not sure what you mean, joined to what?

Reply:
Yeah I think Nick is correct - because Panorama uses this stupid string-based way of passing this data to panels, you need a comma between each string so you can parse them out again - as you're doing in `speedometer-select.js` with
```js
.GetAttributeString('speedometerNames', '').split(',')
```

You have an array of objects of the form `[{ name: foo }, { name: bar }, ... ]`, `.map(x => x.name)` correctly maps that to `[foo, bar, ...]`, but then you need to join those array items together into a string. `.map(x => x.name).join('\')` should be all you need.

Reply:
ok, I understand. The changes have been made to join them into a string. Any reason why it was working fine without sending the data correctly?

Reply:
Were you only testing with one element? If so no joining would be needed, and the splitting would only be an array with one element.

Reply:
Oh, playing around with it in Node repl and ['a', 'b'].toString()` returns `a,b`, or ``${['a', 'b']}`. So `Array.prototype.toString` defaults to comma separators. Still, certainly better to use `.join(',')` for clarity.